File: fixtures/scale/elixir/large/lib/core/store.ex
defmodule Core.Store do
  alias Core.Config

  def save(%Config{results_path: path}, results) do
    File.write(path, Enum.map_join(results, "\n", & &1.subject))
  end

  def load(%Config{results_path: path}) do
    case File.read(path) do
      {:ok, body} -> String.split(body, "\n", trim: true)
      {:error, _reason} -> []
    end
  end

  def append(%Config{results_path: path}, entry) do
    File.write(path, entry <> "\n", [:append])
  end
end

File: fixtures/scale/haskell/large/Core/Store.hs
module Core.Store (save, load, append) where

import Core.Config (Config(..))
import Support.Result (TestResult(..))

save :: Config -> [TestResult] -> IO ()
save cfg rs = writeFile (resultsPath cfg) (unlines (map subject rs))

load :: Config -> IO [String]
load cfg = fmap lines (readFile (resultsPath cfg))

append :: Config -> String -> IO ()
append cfg entry = appendFile (resultsPath cfg) (entry ++ "\n")
